### PR TITLE
feat:H1116-L3 - Refund and sync for Bambora

### PR DESCRIPTION
**Pull request by hey-ritikgupta**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by jarnura — COMMENTED** [review]
Refund related flows are in to-do's, complete the to-do's

**Comment by hey-ritikgupta** [comment]
> Refund related flows are in to-do's, complete the to-do's

Done

File: crates/router/src/connector/bambora.rs [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+        .try_into()
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
+    }
+
+    // fn get_error_response(
+    //     &self,
+    //     res: Response,
+    // ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
+    //     self.build_error_response(res)
+    // }
```
Comment (jagan-jaya) [inline crates/router/src/connector/bambora.rs:311]:
```suggestion
```
remove this commented block

File: crates/router/src/connector/bambora/transformers.rs [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    }
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for BamboraPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let payment_method_data = item.request.payment_method_data.clone();
+        let payment_method = match payment_method_data {
+            api::PaymentMethod::Card(ref _item) => String::from("card"),
+            _ => todo!(),
```
Comment (jagan-jaya) [inline crates/router/src/connector/bambora/transformers.rs:31]: throw proper error message as mentioned in previous PR

File: crates/router/src/connector/bambora/transformers.rs [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        let card = match payment_method_data {
+            api::PaymentMethod::Card(ref item) => Card {
+                name: item.card_holder_name.peek().clone(),
+                number: item.card_number.peek().clone(),
+                expiry_month: item.card_exp_month.peek().clone(),
+                expiry_year: item.card_exp_year.peek().clone(),
+                cvd: item.card_cvc.peek().clone(),
+                complete: true,
+            },
+            _ => todo!(),
```
Comment (jagan-jaya) [inline crates/router/src/connector/bambora/transformers.rs:42]: throw proper error message as mentioned in previous PR

File: crates/router/tests/connectors/bambora.rs [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+        .unwrap();
+    assert_eq!(
+        response.response.unwrap_err().message,
+        "Refund amount (₹1.50) is greater than charge amount (₹1.00)",
+    );
+}
+
+// Connector dependent test cases goes here
+
+// [#478]: add unit tests for non 3DS, wallets & webhooks in connector tests
```
Comment (jagan-jaya) [inline crates/router/tests/connectors/bambora.rs:438]: remove above comments

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

**Comment by ArjunKarthik** [comment]
Thanks for the contribution @hey-ritikgupta , We are merging the changes from different participant as it has more flows implemented. You can still continue your contributions on the [open issues](https://github.com/juspay/hyperswitch/issues?q=is%3Aissue+is%3Aopen+label%3AA-connector-integration).

**Pull request closed without merging**